frontend/src/__tests__/cypress/cypress/pages/clusterStorage.ts

```diff
@@ -0,0 +1,126 @@
+import { Modal } from './components/Modal';
+import { TableRow } from './components/table';
+
+class ClusterStorageRow extends TableRow {
+  shouldHaveStorageTypeValue(name: string) {
+    this.find().find(`[data-label=Type]`).contains(name).should('exist');
+    return this;
+  }
+
+  findConnectedWorkbenches() {
+    return this.find().find('[data-label="Connected workbenches"]');
+  }
+
+  toggleExpandByIndex(index: number) {
+    this.find().find('button').should('have.attr', 'aria-label', 'Details').eq(index).click();
```

Comment:
I believe you can use `findByRole` and `name` for this selector. Also, why do you need to check the `index` ? Shouldn't there only be one per row?
Ideally we use test
```suggestion
    this.find().findByRole('button', { name: 'Details' }).click();
```